### PR TITLE
fix(network_drop_package): Process was starving

### LINKED ISSUE #3135: network_drop_package.py is flaky: times out
A regular run takes 20 seconds, and the timeout is three minutes, so the timeouts are not due to the timeout being too small.

http://52.149.162.182:3000/#/test/7273
http://52.149.162.182:3000/#/test/4873

### PR DESCRIPTION
**Pull request by mfornet**
Add time.sleep(0) to allow other process to make progress.
Reproducing the issue is hard, so there is a chance this doesn't
fix it. In that case it is possible to use time.sleep(eps) but it
leads to a suboptimal experience where we need to wait `eps` timeout
for every message passing through the proxy.

Fixes #3135 

Test plan
=========
network_drop_package.py should pass

Success. http://nayduck.eastus.cloudapp.azure.com:3000/run/214

### CODE REVIEW

**Comment by mfornet** [comment]
> Please try running it multiple times. You can submit 50 runs to nayduck to see if it is fixed

Running here: http://nayduck.eastus.cloudapp.azure.com:3000/run/151#/

**Comment by bowenwang1996** [comment]
@mfornet looks like it still fails sometimes

**Comment by SkidanovAlex** [comment]
It seems to fail with a different reason, so likely Marcelo fixed the issue that was appearing before (I would resubmit the whole run, because nayduck was under maintenance at the time of that run, potentially causing some of the issues?).

But also what's weird is that `stress.py` with `packets_drop` mode does the same thing, but with significantly more load, and it hasn't failed the same way for me.

@mfornet can you resubmit nayduck run keeping the part ` and not finished` (that ensures that the success value is only set once), but removing the sleep, and also tracing at the end of the main process (so that we can differentiate between the situation when the main process never breaks vs when it breaks, but the test doesn't stop because other processes keep running)?

**Comment by mfornet** [comment]
I can reproduce locally the behavior from nayduck. On some runs the main process never breaks, it just doesn't make any progress in the last loop, while other process are making progress.

What is more weird, is that depending on the position of the logging (a print call) it actually makes progress or not.

File: pytest/lib/proxy.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def other(self, ordinal_a, ordinal_b):
             return ordinal_a
 
     async def _handle(self, raw_message, *, writer, sender_port_holder, receiver_port_holder, ordinal_to_writer):
-        # time.sleep(0)
+        time.sleep(1e-6)
```
[the remark this answers is not in the document]
Comment (mfornet) [reply pytest/lib/proxy.py]: This was removed already

**Comment by mfornet** [comment]
nayduck is green: http://nayduck.eastus.cloudapp.azure.com:3000/run/214#/run/214

File: pytest/tests/sanity/network_drop_package.py [hunk 2]
```diff
@@ -16,26 +17,30 @@
 # Ratio of message that are dropped to simulate bad network performance
 DROP_RATIO = 0.05
 
+DONT_FILTER = ['Handshake', 'HandshakeFailure', 'LastEdge', 'Sync', 'RequestUpdateNonce', 'ResponseUpdateNonce']
```
Comment (SkidanovAlex) [inline pytest/tests/sanity/network_drop_package.py:20]: In stress.py packages_drop I only filter Handshake*, and it works fine. Are you sure you need to filter all of these?

**Pull request merged**